src/bin/stratisd.rs

```diff
@@ -83,10 +105,23 @@ fn run() -> StratisResult<()> {
         }
     };
 
-    let (dbus_conn, mut tree, dbus_context) = libstratis::dbus_api::connect(Rc::clone(&engine))?;
+    let (mut dbus_conn, mut tree, base_object_path, mut dbus_context) =
```

Comment:
It was pretty messy when ```fds``` was shared by only two different types of file descriptors requiring special treatment. If it's going to be shared by 3, which it probably has to be, then this whole setup should get some improvement.

Reply:
#656 is for this.